snap/snapcraft.yaml

```diff
@@ -0,0 +1,54 @@
+name: starcraft
+version: git
+base: core24
+summary: Single-line elevator pitch for your amazing snap # 79 char long summary
+description: |
+  This is starcraft description. You have a paragraph or two to tell the
+  most important story about your snap. Keep it under 100 words though,
+  we live in tweetspace and your description wants to look good in the snap
+  store.
+confinement: strict
```

Comment:
I wonder whether we should just put classic here since all of our snaps currently are classic...

Reply:
I wondered too but then saw that our newest app is `strict` and was able to get away with some plugs. So maybe we should keep `strict` by default and have the action to set another confinement a conscious decision.

Reply:
Or does a `snapcraft.yaml` boilerplate that would work for our apps belong in https://github.com/canonical/craftcraft? It already has one, but is missing recent updates like the parts for `libgit2`, `git`, `chisel`, and the auto-completer.

I'm fine to have starbase's snapcraft.yaml be the absolute minimum to snap a UV project. I'd even be fine to drop the craft-cli autocompleter.

Reply:
Now that craftcraft was removed, starbase is the next best home but I'm still not convinced it belongs here.

I'm still unsure of the value proposition of this snapcraft.yaml since it's somewhere in between a generic `uv` application and a craft app.

Reply:
This semi-generic snapcraft.yaml serves several purposes:
- when starting a new craft application, it makes it easier to get a basic snap. Without this file users have to write one from scratch.
- propose a template implementing as much as possible what we consider the basic features of our snaps (auto-completion, platforms list, license, etc.)
- help spot differences in our `snapcraft.yaml` when rebasing on starbase and so help harmonize them.